gov/politeia/proposals.go

```diff
@@ -90,371 +87,509 @@ func NewProposalsDB(politeiaURL, dbPath string) (*ProposalDB, error) {
+	recordDetails, err := db.fetchRecordDetails(tokens)
 	if err != nil {
-		log.Errorf("Failed to fetch data from Proposals DB: %v", err)
 		return nil, err
 	}
 
-	return &pInfo, nil
-}
+	// Fetch comments count for each token from the inventory
+	cr, err := db.client.CommentCount(commentsv1.Count{
+		Tokens: tokens,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Pi client CommentCount err: %v", err)
+	}
+	commentsCounts := cr.Counts
 
-// LastProposalsSync returns the last time a sync to update the proposals was run
-// but not necessarily the last time updates were synced in proposals.db.
-func (db *ProposalDB) LastProposalsSync() int64 {
-	return atomic.LoadInt64(&db.lastSync)
-}
+	// Fetch vote summary for each token from the inventory
+	sr, err := db.client.TicketVoteSummaries(ticketvotev1.Summaries{
+		Tokens: tokens,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Pi client TicketVoteSummaries err: %v", err)
+	}
+	voteSummaries := sr.Summaries
 
-// CheckProposalsUpdates updates the proposal changes if they exist and updates
-// them to the proposal db.
-func (db *ProposalDB) CheckProposalsUpdates() error {
-	if db == nil || db.dbP == nil {
-		return errDef
+	var proposals []*pitypes.ProposalRecord
+
+	// Iterate through every record and feed data used by dcrdata
+	for _, record := range recordDetails {
+		proposal := pitypes.ProposalRecord{}
+
+		// Record data
+		proposal.State = record.State
+		proposal.Status = record.Status
+		proposal.Version = record.Version
+		proposal.Timestamp = uint64(record.Timestamp)
+		proposal.Username = record.Username
+		proposal.Token = record.CensorshipRecord.Token
+
+		// Proposal metadata
+		pm, err := proposalMetadataDecode(record.Files)
+		if err != nil {
+			return nil, fmt.Errorf("proposalMetadataDecode err: %v", err)
+		}
+		proposal.Name = pm.Name
+
+		// User metadata
+		um, err := userMetadataDecode(record.Metadata)
+		if err != nil {
+			return nil, fmt.Errorf("userMetadataDecode err: %v", err)
+		}
+		proposal.UserID = um.UserID
+
+		// Comments count
+		proposal.CommentsCount =
+			int32(commentsCounts[record.CensorshipRecord.Token])
+
+		// Vote summary data
+		summary := voteSummaries[proposal.Token]
+		proposal.VoteStatus = summary.Status
+		proposal.VoteResults = summary.Results
+		proposal.EligibleTickets = summary.EligibleTickets
+		proposal.StartBlockHeight = summary.StartBlockHeight
+		proposal.EndBlockHeight = summary.EndBlockHeight
+		proposal.QuorumPercentage = summary.QuorumPercentage
+		proposal.PassPercentage = summary.PassPercentage
+
+		var totalVotes uint64
+		for _, v := range summary.Results {
+			totalVotes += v.Votes
+		}
+		proposal.TotalVotes = totalVotes
+
+		// Status change metadata
+		ts, changeMsg, err := statusChangeMetadataDecode(record.Metadata)
+		if err != nil {
+			return nil, fmt.Errorf("statusChangeMetadataDecode err: %v", err)
+		}
+		proposal.PublishedAt = ts[0]
+		proposal.CensoredAt = ts[1]
+		proposal.AbandonedAt = ts[2]
```

Comment:
It would be clearer and less error prone for `statusChangeMetadataDecode` to return something like


```go
type statusTime struct {
    published int64
    censored int64
    abandoned int64
}
```

and then 

```go
proposal.PublishedAt = times.published
proposal.CensoredAt = times.censored
proposal.AbandonedAt = times.abandoned
```